uint/src/uint.rs

```diff
@@ -1550,31 +1550,34 @@ macro_rules! construct_uint {
 			}
 		}
 
-		$crate::impl_std_for_uint!($name, $n_words);
-		// `$n_words * 8` because macro expects bytes and
-		// uints use 64 bit (8 byte) words
-		$crate::impl_quickcheck_arbitrary_for_uint!($name, ($n_words * 8));
-		$crate::impl_arbitrary_for_uint!($name, ($n_words * 8));
-	}
-}
-
-#[cfg(feature = "std")]
-#[macro_export]
-#[doc(hidden)]
-macro_rules! impl_std_for_uint {
-	($name: ident, $n_words: tt) => {
 		impl $crate::core_::str::FromStr for $name {
-			type Err = $crate::rustc_hex::FromHexError;
+			type Err = $crate::hex::FromHexError;
 
 			fn from_str(value: &str) -> $crate::core_::result::Result<$name, Self::Err> {
-				use $crate::rustc_hex::FromHex;
-				let bytes: Vec<u8> = match value.len() % 2 == 0 {
-					true => value.from_hex()?,
-					false => ("0".to_owned() + value).from_hex()?,
-				};
+				const bytes_len: usize = $n_words * 8;
+				const max_encoded_len: usize = bytes_len * 2;
+
+				let mut bytes = [0_u8; bytes_len];
+
+				let encoded = value.trim_start_matches('0').as_bytes();
```

Comment:
and that changes the semantics allowing arbitrary number of leading zeroes

Reply:
I could omit this entirely, I suppose? Although IMO trimming is the right thing to do, as the number with a lot of leading zeroes should still be legal integer.

Reply:
Removed trimming from PR